modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/SegmentsExperienceModelListenerTest.java

```diff
@@ -0,0 +1,191 @@
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getCompanyId(), _group.getGroupId(),
+			TestPropsValues.getUserId());
+
+		_layout = LayoutTestUtil.addTypeContentLayout(_group);
+	}
+
+	@Test
+	public void testOnAfterUpdate() throws Exception {
+		long defaultSegmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				_layout.getPlid());
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.addSegmentsExperiment(
+				defaultSegmentsExperienceId,
+				_portal.getClassNameId(Layout.class), _layout.getPlid(),
+				"AB test", "A/B test description",
+				SegmentsExperimentConstants.Goal.BOUNCE_RATE.getLabel(),
+				StringPool.BLANK, _serviceContext);
+
+		SegmentsExperience segmentsExperience1 =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				_portal.getClassNameId(Layout.class), _layout.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Variant 1"),
+				false, new UnicodeProperties(true), _serviceContext);
+
+		_segmentsExperimentRelService.addSegmentsExperimentRel(
+			segmentsExperiment.getSegmentsExperimentId(),
+			segmentsExperience1.getSegmentsExperienceId(), _serviceContext);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment.getSegmentsExperienceId(),
+					() -> "{}")));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String name = "Variant 1 modified";
+
+			segmentsExperience1.setNameMap(
+				Collections.singletonMap(LocaleUtil.getSiteDefault(), name));
+
+			_segmentsExperienceLocalService.updateSegmentsExperience(
+				segmentsExperience1);
+
+			SegmentsExperience segmentsExperience2 =
+				_segmentsExperienceLocalService.getSegmentsExperience(
+					segmentsExperience1.getSegmentsExperienceId());
+
```

Comment:
Here you are not asserting the update call to AC, so maybe this test should be translated into a unit test or not tested at all.

Reply:
This class has been deleted and I added some unit tests of AsahFaroBackendClientImplTest class.